include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -144,81 +204,244 @@ class is_operational_impl
-            output_bdl_wires{output_wires}
+            output_bdl_wires{output_wires},
+            number_of_output_wires{output_bdl_wires.size()},
+            number_of_input_wires{input_bdl_wires.size()},
+            canvas_lyt{c_lyt}
+    {
+        if (params.op_condition == is_operational_params::operational_condition::TOLERATE_KINKS)
+        {
+            output_bdl_pairs = detect_bdl_pairs(layout, sidb_technology::cell_type::OUTPUT,
+                                                params.input_bdl_iterator_params.bdl_wire_params.bdl_pairs_params);
+        }
+        canvas_lyt.foreach_cell(
+            [this](const auto& c)
+            {
+                dependent_cell = c;
+                return false;
+            });
+    }
+
+    /**
+     * Constructor to initialize the algorithm with a layout and parameters.
+     *
+     * @param lyt The SiDB cell-level layout to be checked.
+     * @param spec Expected Boolean function of the layout given as a multi-output truth table.
+     * @param params Parameters for the `is_operational` algorithm.
+     */
+    is_operational_impl(const Lyt& lyt, const std::vector<TT>& tt, const is_operational_params& params,
+                        const Lyt& c_lyt) :
+            layout{lyt},
+            truth_table{tt},
+            parameters{params},
+            output_bdl_pairs(detect_bdl_pairs(lyt, sidb_technology::cell_type::OUTPUT,
+                                              params.input_bdl_iterator_params.bdl_wire_params.bdl_pairs_params)),
+            bii(bdl_input_iterator<Lyt>{lyt, params.input_bdl_iterator_params}),
+            input_bdl_wires{
+                detect_bdl_wires(lyt, params.input_bdl_iterator_params.bdl_wire_params, bdl_wire_selection::INPUT)},
+            output_bdl_wires{
+                detect_bdl_wires(lyt, params.input_bdl_iterator_params.bdl_wire_params, bdl_wire_selection::OUTPUT)},
+            number_of_output_wires{output_bdl_wires.size()},
+            number_of_input_wires{input_bdl_wires.size()},
+            canvas_lyt{c_lyt}
     {}
 
+    /**
+     * This function evaluates whether the given layout can be discarded since it cannot implement the given Boolean
+     * function. The filtering is subdivided into three single filtering steps: (1) discarding SiDB layouts with
+     * potentially positively charged SiDBs, (2) utilizing an efficient method to identify and discard SiDB layouts that
+     * do not satisfy physical model constraints under the I/O pin conditions required for the desired Boolean function,
+     * and (3) detecting I/O signal instability.
+     *
+     * @tparam ChargeLyt The charge distribution surface layout type.
+     * @param input_pattern The current input pattern.
+     * @param cds_canvas The charge distribution of the canvas layout.
+     * @param dependent_cell A dependent-cell of the canvas SiDBs.
+     * @return An optional layout_invalidity_reason indicating the reason why the layout is non-operational.
```

Comment:
```suggestion
     * @return A `layout_invalidity_reason` object indicating why the layout is non-operational; or `std::nullopt` if it could not certainly be determined to be in fact non-operational.
```